### PR TITLE
Fulton Rework

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
balance: Fultons now work no matter your location.
balance: Fultons now work on living mobs. He's coming too?
balance: Fultons now no longer have a use limit.
balance: Fultons now have travel time, 10 seconds for every 100 tiles, this scales based on the amount of tiles.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
how does it travel instation?

**Comment by fludd12** [comment]
They go through the ceiling, of course.

**Comment by JJRcop** [comment]
I think this PR should be retitled from rework to rebalance, since the changes are so light.

**Comment by Xhuis** [comment]
There is no ceiling. All the z-levels are stacked on top of each other with insulating spacetime weave between them preventing air from moving flow, like a layered cake.

File: code/modules/mining/fulton.dm [hunk 1]
```diff
@@ -109,10 +91,13 @@ GLOBAL_LIST_EMPTY(total_extraction_beacons)
 				L.SetUnconscious(0)
 				L.drowsyness = 0
 				L.SetSleeping(0)
-			sleep(30)
 			var/list/flooring_near_beacon = list()
 			for(var/turf/open/floor in orange(1, beacon))
 				flooring_near_beacon += floor
+			if(beacon.z == z)
+				sleep(10 * (get_dist(get_turf(src), get_turf(beacon)) / 100)
```
Comment (JJRcop) [inline code/modules/mining/fulton.dm]: didn't close the sleep call

**Comment by ExcessiveUseOfCobblestone** [comment]
So when you're ""traveling"" you're really just on the same turf but stunned and invisible?

File: code/modules/mining/fulton.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/modules/mining/fulton.dm]: goddamn it

**Comment by Iamgoofball** [comment]
@ExcessiveUseOfCobblestone Yeah, but you're hidden from view and unable to be clicked.

**Comment by Iamgoofball** [comment]
@fludd12 What ceiling?

**Comment by MMMiracles** [comment]
Using a fulton in a station area should cause the sound of someone crashing through some metal and generic clutter/rubble being spawned around the tile when they fly up.

**Comment by CameronWoof** [comment]
Security will love this. Free station-wide prisoner teleportation? Yes please.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Iamgoofball what about npcs :o ??

**Comment by Iamgoofball** [comment]
@mmmiracles there's no ceiling 

@cobby yes

**Comment by ExcessiveUseOfCobblestone** [comment]
THATS WHY I DIED ONCE AHHHHHHHHHHHHHHHHHHHHHHHHH

**Comment by Robustin** [comment]
Uh fultons already work on living mobs.

I often fulton myself back to station. 

@CameronWoof 

I've tried to give engineer access to indoor fultons before but it was a simple var-change that Goof had already coded.

**Comment by Dorsisdwarf** [comment]
can you fulton between z levels? I forget.

if so,  how does this calculate distance between two locations on different z levels

**Comment by Iamgoofball** [comment]
if it's cross z-levels it just takes 10 seconds

**Comment by Doppeltongue** [comment]
Fultons have always worked with people, what are you smoking, goof?

**Comment by Iamgoofball** [comment]
@Doppeltongue There was an option to disable it, I removed it.

**Comment by MrDoomBringer** [comment]
image

**Comment by kevinz000** [comment]
they should not be crossing zlevels and infinite use is a bit :/

**Comment by kevinz000** [comment]
also >upthumbing your own pr

**Comment by 81Denton** [comment]
>they should not be crossing zlevels

But why

**Comment by 81Denton** [comment]
Good PR, this might give fultons some more use. According to statbus they're less popular than crushers and even mining drones.

**Comment by Iamgoofball** [comment]
Z-level crossing is not an issue in the slightest.

**Comment by ExcessiveUseOfCobblestone** [comment]
They've always had z-level crossing? I've fultoned myself to cargo before.

**Comment by Iamgoofball** [comment]
^

**Comment by Iamgoofball** [comment]
My GPU is fucking dead, this is on hold indefinitely

**Pull request closed without merging**